### PR TITLE
Make Nip46 Magical Again

### PR DESCRIPTION
**Pull request by pablof7z**
This PR implements the magical changes of NIP-46 I'm going to add soon

### CODE REVIEW

File: ndk/src/signers/nip46/backend/nip04-encrypt.ts [hunk 1]
```diff
@@ -4,13 +4,29 @@ import type { IEventHandlingStrategy, NDKNip46Backend } from "./index.js";
...
     async handle(
         backend: NDKNip46Backend,
+        id: string,
         remotePubkey: string,
         params: string[]
     ): Promise<string | undefined> {
         const [recipientPubkey, payload] = params;
         const recipientUser = new NDKUser({ hexpubkey: recipientPubkey });
-        const decryptedPayload = await backend.encrypt(remotePubkey, recipientUser, payload);
+        const decryptedPayload = await encrypt(backend, id, remotePubkey, recipientUser, payload);
```
Comment (erskingardner) [inline ndk/src/signers/nip46/backend/nip04-encrypt.ts:13]: This should be `const encryptedPayload` right?

File: ndk/src/signers/nip46/backend/nip04-encrypt.ts [hunk 2]
```diff
@@ -4,13 +4,29 @@ import type { IEventHandlingStrategy, NDKNip46Backend } from "./index.js";
...
 
         return decryptedPayload;
```
Comment (erskingardner) [inline ndk/src/signers/nip46/backend/nip04-encrypt.ts:15]: here too.

Comment (erskingardner) [reply ndk/src/signers/nip46/backend/nip04-encrypt.ts:15]: also, in the decrypt method you're returning the serialized value using `JSON.stringify` do you want to do that here too?

File: ndk/src/signers/nip46/rpc.ts [hunk 3]
```diff
@@ -117,7 +121,16 @@ export class NDKNostrRpc extends EventEmitter {
         const remoteUser = this.ndk.getUser({ hexpubkey: remotePubkey });
         const request = { id, method, params };
         const promise = new Promise<NDKRpcResponse>((resolve) => {
-            if (cb) this.once(`response-${id}`, cb);
+            const responseHandler = (response: NDKRpcResponse) => {
```
Comment (erskingardner) [inline ndk/src/signers/nip46/rpc.ts:124]: I don't see anything dick-buttish in here but I'm not exactly sure what this `auth_url` flow is so I'm not 100% on this...

Comment (pablof7z) [reply ndk/src/signers/nip46/rpc.ts:124]: this is the magical part

File: ndk/src/signers/nip46/index.ts [hunk 4]
```diff
@@ -9,13 +10,18 @@ import { NDKNostrRpc } from "./rpc.js";
 /**
  * This NDKSigner implements NIP-46, which allows remote signing of events.
  * This class is meant to be used client-side, paired with the NDKNip46Backend or a NIP-46 backend (like Nostr-Connect)
+ *
+ * @emits authUrl -- Emitted when the user should take an action in certain URL.
```
Comment (v0l) [inline ndk/src/signers/nip46/index.ts:14]: like "sign in with <platform>" to access key with bunker?

**Pull request merged**